### DISCUSSION TITLE
can't build an heltec_wireless_stick

### DISCUSSION THREAD

**Discussion by teufti** [Velocet/iown-homecontrol#29] [General / Q&A]
Hello

with vscode, i want to build heltec_wireless_stick V3 : 
in platformio.ini i added : 
[env:heltec_wireless_stick]
board       = heltec_wireless_stick
extends     = platform_espressif32
board_url   = https://resource.heltec.cn/download/Wireless_Stick_V3

And when i build , there is this message in the terminal : 

*  Executing task: platformio run --target upload --environment heltec_wireless_stick 

Processing heltec_wireless_stick (board: heltec_wireless_stick; platform: espressif32; framework: arduino)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/heltec_wireless_stick.html
PLATFORM: Espressif 32 (6.7.0) > Heltec Wireless Stick
HARDWARE: ESP32 240MHz, 320KB RAM, 8MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 3.20016.0 (2.0.16) 
 - tool-esptoolpy @ 1.40501.0 (4.5.1) 
 - tool-mkfatfs @ 2.0.1 
 - tool-mklittlefs @ 1.203.210628 (2.3) 
 - tool-mkspiffs @ 2.230.0 (2.30) 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ off
Found 35 compatible libraries
Scanning dependencies...
Dependency Graph
|-- SPI @ 2.0.0
|-- RadioLib @ 6.6.0+sha.bc36c1e
|-- LoRa32 @ 0.6.9+sha.4d4a167
Building in release mode
Linking .pio/build/heltec_wireless_stick/firmware.elf
/home/pat/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pio/build/heltec_wireless_stick/src/main_IoHome.cpp.o: in function `dummyISR()':
/home/pat/PlatformIO/Projects/iown-homecontrol/src/main_IoHome.cpp:26: multiple definition of `dummyISR()'; .pio/build/heltec_wireless_stick/src/main.cpp.o:/home/pat/PlatformIO/Projects/iown-homecontrol/src/main.cpp:28: first defined here
/home/pat/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pio/build/heltec_wireless_stick/src/main_IoHome.cpp.o:/home/pat/PlatformIO/Projects/iown-homecontrol/src/main_IoHome.cpp:19: multiple definition of `radio'; .pio/build/heltec_wireless_stick/src/main.cpp.o:/home/pat/PlatformIO/Projects/iown-homecontrol/src/main.cpp:21: first defined here
/home/pat/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pio/build/heltec_wireless_stick/src/main_IoHome.cpp.o:/home/pat/PlatformIO/Projects/iown-homecontrol/src/main_IoHome.cpp:22: multiple definition of `phy'; .pio/build/heltec_wireless_stick/src/main.cpp.o:/home/pat/PlatformIO/Projects/iown-homecontrol/src/main.cpp:24: first defined here
/home/pat/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pio/build/heltec_wireless_stick/src/main_IoHome.cpp.o: in function `setup()':
/home/pat/PlatformIO/Projects/iown-homecontrol/src/main_IoHome.cpp:28: multiple definition of `setup()'; .pio/build/heltec_wireless_stick/src/main.cpp.o:/home/pat/PlatformIO/Projects/iown-homecontrol/src/main.cpp:30: first defined here
/home/pat/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pio/build/heltec_wireless_stick/src/main_IoHome.cpp.o: in function `loop()':
/home/pat/PlatformIO/Projects/iown-homecontrol/src/main_IoHome.cpp:56: multiple definition of `loop()'; .pio/build/heltec_wireless_stick/src/main.cpp.o:/home/pat/PlatformIO/Projects/iown-homecontrol/src/main.cpp:58: first defined here
collect2: error: ld returned 1 exit status
*** [.pio/build/heltec_wireless_stick/firmware.elf] Error 1
====================================================================== [FAILED] Took 8.67 seconds ======================================================================

Environment            Status    Duration
---------------------  --------  ------------
heltec_wireless_stick  FAILED    00:00:08.667 

why build failed ?

Thanks for your help

**Comment by eriknn** [comment]
I believe the code contains an old outdated code as well as never code (?). This gives you two "main"-loops.

I recently stumbled over this project, so I don't really know what's going on, but if you delete main_IoHome.cpp (and probably the other 2 IoHome-files), you'll be able to build!

**Comment by teufti** [comment]
I removed the 3 files, now the build runs successfully, but the upload ends with the following error:

A fatal error occurred: This chip is ESP32-S3 not ESP32. Wrong --chip argument?
*** [upload] Error 2
================================================================== [FAILED] Took 6.28 seconds ==================================================================

Environment             Status    Duration
----------------------  --------  ------------
heltec_wifi_lora_32_V2  FAILED    00:00:06.283
============================================================ 1 failed, 0 succeeded in 00:00:06.283 ============================================================

 *  The terminal process "platformio 'run', '--target', 'upload', '--environment', 'heltec_wifi_lora_32_V2'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

Thank you Best regards

**Comment by eriknn** [comment]
You are using a different ESP than the code was made for. You must change to ESP-s3 in platformio.ini. After this you will get some other problems as well.
